File: Server/aoRss/Controllers/GenericController.vb

Imports Contensive.BaseClasses

Namespace Controllers
    Public NotInheritable Class GenericController
        '
        '
        Public Shared Function GetGMTFromDate(DateValue As Date) As String
            Dim result As String = ""
            Dim WorkLong As Integer
            '
            If IsDate(DateValue) Then
                Select Case Weekday(DateValue)
                    Case vbSunday
                        result = "Sun, "
                    Case vbMonday
                        result = "Mon, "
                    Case vbTuesday
                        result = "Tue, "
                    Case vbWednesday
                        result = "Wed, "
                    Case vbThursday
                        result = "Thu, "
                    Case vbFriday
                        result = "Fri, "
                    Case vbSaturday
                        result = "Sat, "
                End Select
                '
                WorkLong = Day(DateValue)
                If WorkLong < 10 Then
                    result = result & "0" & CStr(WorkLong) & " "
                Else
                    result = result & CStr(WorkLong) & " "
                End If
                '
                Select Case Month(DateValue)
                    Case 1
                        result = result & "Jan "
                    Case 2
                        result = result & "Feb "
                    Case 3
                        result = result & "Mar "
                    Case 4
                        result = result & "Apr "
                    Case 5
                        result = result & "May "
                    Case 6
                        result = result & "Jun "
                    Case 7
                        result = result & "Jul "
                    Case 8
                        result = result & "Aug "
                    Case 9
                        result = result & "Sep "
                    Case 10
                        result = result & "Oct "
                    Case 11
                        result = result & "Nov "
                    Case 12
                        result = result & "Dec "
                End Select
                '
                result = result & CStr(Year(DateValue)) & " "
                '
                WorkLong = Hour(DateValue)
                If WorkLong < 10 Then
                    result = result & "0" & CStr(WorkLong) & ":"
                Else
                    result = result & CStr(WorkLong) & ":"
                End If
                '
                WorkLong = Minute(DateValue)
                If WorkLong < 10 Then
                    result = result & "0" & CStr(WorkLong) & ":"
                Else
                    result = result & CStr(WorkLong) & ":"
                End If
                '
                WorkLong = Second(DateValue)
                If WorkLong < 10 Then
                    result = result & "0" & CStr(WorkLong)
                Else
                    result = result & CStr(WorkLong)
                End If
                '
                result = result & " GMT"
            End If
            Return result
        End Function
        '
        '====================================================================================================
        ''' <summary>
        ''' return an anchor tag for the provided feed
        ''' </summary>
        ''' <param name="cp"></param>
        ''' <param name="rssfeed"></param>
        ''' <returns></returns>
        Public Shared Function getFeedLink(cp As CPBaseClass, rssfeed As Models.Db.RSSFeedModel) As String
            If rssfeed.RSSFilename = "" Then
                Return "<img src=""/rssFeeds/IconXML-25x13.gif"" width=25 height=13 border=0 class=""RSSFeedImage"">&nbsp;" & rssfeed.name & "&nbsp;(Coming Soon)"
            Else
                Return "<img src=""/rssFeeds/IconXML-25x13.gif"" width=25 height=13 border=0 class=""RSSFeedImage"">&nbsp;<a class=""RSSFeedLink"" href=""" & cp.Http.CdnFilePathPrefixAbsolute & rssfeed.RSSFilename & """>" & rssfeed.name & "</a>"
            End If
        End Function
        '
        '====================================================================================================
        ''' <summary>
        ''' tmp Placeholder for CP.Http.CdnFilePathPrefixAbsolute
        ''' </summary>
        ''' <param name="cp"></param>
        ''' <returns></returns>
        Public Shared Function getCdnFilePathPrefixAbsolute(cp As CPBaseClass) As String

            If (Not cp.ServerConfig.isLocalFileSystem) Then
                '
                ' -- remote file system, return cdnfileurl
                Return cp.GetAppConfig().cdnFileUrl
            Else
                '
                ' -- local file system
                Dim cdnFilePathPrefixAbsolute As String = cp.Site.GetText("CdnFilePathPrefixAbsolute")?.Replace("\", "/")
                If (Not String.IsNullOrWhiteSpace(cdnFilePathPrefixAbsolute)) Then
                    If (Not cdnFilePathPrefixAbsolute.Substring(cdnFilePathPrefixAbsolute.Length, 1).Equals("/")) Then
                        cdnFilePathPrefixAbsolute &= "/"
                    End If
                Else
                    cdnFilePathPrefixAbsolute = "https://" & cp.Site.DomainPrimary & "/" & cp.GetAppConfig().cdnFileUrl
                End If
                Return cdnFilePathPrefixAbsolute
            End If
        End Function

    End Class
End Namespace

File: Server/aoRss/Controllers/GenericController.cs
using System;
using Contensive.BaseClasses;

namespace Contensive.Addons.Rss.Controllers {
    public sealed class GenericController {
        //
        //
        public static string GetGMTFromDate(DateTime DateValue) {
            string result = "";
            int WorkLong;
            //
            if (DateValue != default(DateTime)) {
                switch (DateValue.DayOfWeek) {
                    case DayOfWeek.Sunday:
                        result = "Sun, ";
                        break;
                    case DayOfWeek.Monday:
                        result = "Mon, ";
                        break;
                    case DayOfWeek.Tuesday:
                        result = "Tue, ";
                        break;
                    case DayOfWeek.Wednesday:
                        result = "Wed, ";
                        break;
                    case DayOfWeek.Thursday:
                        result = "Thu, ";
                        break;
                    case DayOfWeek.Friday:
                        result = "Fri, ";
                        break;
                    case DayOfWeek.Saturday:
                        result = "Sat, ";
                        break;
                }
                //
                WorkLong = DateValue.Day;
                if (WorkLong < 10) {
                    result = $"{result}0{WorkLong} ";
                } else {
                    result = $"{result}{WorkLong} ";
                }
                //
                switch (DateValue.Month) {
                    case 1:
                        result = $"{result}Jan ";
                        break;
                    case 2:
                        result = $"{result}Feb ";
                        break;
                    case 3:
                        result = $"{result}Mar ";
                        break;
                    case 4:
                        result = $"{result}Apr ";
                        break;
                    case 5:
                        result = $"{result}May ";
                        break;
                    case 6:
                        result = $"{result}Jun ";
                        break;
                    case 7:
                        result = $"{result}Jul ";
                        break;
                    case 8:
                        result = $"{result}Aug ";
                        break;
                    case 9:
                        result = $"{result}Sep ";
                        break;
                    case 10:
                        result = $"{result}Oct ";
                        break;
                    case 11:
                        result = $"{result}Nov ";
                        break;
                    case 12:
                        result = $"{result}Dec ";
                        break;
                }
                //
                result = $"{result}{DateValue.Year} ";
                //
                WorkLong = DateValue.Hour;
                if (WorkLong < 10) {
                    result = $"{result}0{WorkLong}:";
                } else {
                    result = $"{result}{WorkLong}:";
                }
                //
                WorkLong = DateValue.Minute;
                if (WorkLong < 10) {
                    result = $"{result}0{WorkLong}:";
                } else {
                    result = $"{result}{WorkLong}:";
                }
                //
                WorkLong = DateValue.Second;
                if (WorkLong < 10) {
                    result = $"{result}0{WorkLong}";
                } else {
                    result = $"{result}{WorkLong}";
                }
                //
                result = $"{result} GMT";
            }
            return result;
        }
        //
        // ====================================================================================================
        /// <summary>
        /// return an anchor tag for the provided feed
        /// </summary>
        /// <param name="cp"></param>
        /// <param name="rssfeed"></param>
        /// <returns></returns>
        public static string getFeedLink(CPBaseClass cp, Models.Db.RSSFeedModel rssfeed) {
            if (string.IsNullOrEmpty(rssfeed.RSSFilename)) {
                return $"<img src=\"/rssFeeds/IconXML-25x13.gif\" width=25 height=13 border=0 class=\"RSSFeedImage\">&nbsp;{rssfeed.name}&nbsp;(Coming Soon)";
            } else {
                return $"<img src=\"/rssFeeds/IconXML-25x13.gif\" width=25 height=13 border=0 class=\"RSSFeedImage\">&nbsp;<a class=\"RSSFeedLink\" href=\"{cp.Http.CdnFilePathPrefixAbsolute}{rssfeed.RSSFilename}\">{rssfeed.name}</a>";
            }
        }
        //
        // ====================================================================================================
        /// <summary>
        /// tmp Placeholder for CP.Http.CdnFilePathPrefixAbsolute
        /// </summary>
        /// <param name="cp"></param>
        /// <returns></returns>
        public static string getCdnFilePathPrefixAbsolute(CPBaseClass cp) {

            if (!cp.ServerConfig.isLocalFileSystem) {
                //
                // -- remote file system, return cdnfileurl
                return cp.GetAppConfig().cdnFileUrl;
            } else {
                //
                // -- local file system
                string cdnFilePathPrefixAbsolute = cp.Site.GetText("CdnFilePathPrefixAbsolute")?.Replace(@"\", "/");
                if (!string.IsNullOrWhiteSpace(cdnFilePathPrefixAbsolute)) {
                    if (!cdnFilePathPrefixAbsolute.Substring(cdnFilePathPrefixAbsolute.Length, 1).Equals("/")) {
                        cdnFilePathPrefixAbsolute += "/";
                    }
                } else {
                    cdnFilePathPrefixAbsolute = $"https://{cp.Site.DomainPrimary}/{cp.GetAppConfig().cdnFileUrl}";
                }
                return cdnFilePathPrefixAbsolute;
            }
        }

    }
}
